lib/kafka_ex/server.ex

```diff
@@ -367,9 +383,11 @@ defmodule KafkaEx.Server do
         {:noreply, update_metadata(state)}
       end
 
-      def update_metadata(state) do
-        {correlation_id, metadata} = retrieve_metadata(state.brokers, state.correlation_id, config_sync_timeout())
-        metadata_brokers = metadata.brokers
+      def update_metadata(state), do: update_metadata(state, nil)
+
+      def update_metadata(state, api_version) do
+        {correlation_id, metadata} = retrieve_metadata_with_version(state.brokers, state.correlation_id, config_sync_timeout(), api_version)
+        metadata_brokers = metadata.brokers |> Enum.map(&(%{&1 | is_controller: &1.node_id == metadata.controller_id}))
```

Comment:
is this change necessary for you PR can you indicate why?

Reply:
We had to add the `api_version` argument to functions handling metadata because in the [Metadata API](https://kafka.apache.org/protocol.html#The_Messages_Metadata) response object version 0 does not provide the `controller_id` value, which is needed to create a topic.

Therefore, it was necessary to allow support for the version 1 of the Metadata API, hence the need to specify the version in the function call.